src/main/java/com/softserve/demo/controller/ProvidersController.java

```diff
@@ -24,40 +25,44 @@
 @CrossOrigin("*")
 public class ProvidersController {
 
-    @Autowired
-    private ProvidersService providersService;
+    private final ProvidersService providersService;
 
-    @Autowired
-    FilesStorageService fileStorageService;
+    private final FilesStorageService fileStorageService;
+
+
+    public ProvidersController(ProvidersService providersService, FilesStorageService fileStorageService) {
+        this.providersService = providersService;
+        this.fileStorageService = fileStorageService;
+    }
 
     @PostMapping("save")
-    public ResponseEntity<Provider> saveServiceProvider(@RequestBody Provider providers) {
-        return new ResponseEntity<>(providersService.save(providers), HttpStatus.OK);
+    public ResponseEntity<?> saveServiceProvider(@RequestBody ProviderDTO providerDTO) {
+        return new ResponseEntity<>(providersService.save(providerDTO), HttpStatus.OK);
```

Comment:
You response with simple object to the save method. Why ? 1.It`s not a DTO. 2. It`s the same object that we have sent to server.